src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,31 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
+      console.log(
+        '[DEBUG] Payload following call to `with`:\n',
+        currentPayload
+      );
+    }
     return new AnalyticsEventReporter(this.config, currentPayload);
   }
 
   public async report(newPayload?: EventPayload): Promise<string> {
+    if (this.config.debug) {
+      console.log(
+        `[DEBUG] Merging the following payloads:
+
+        Original Payload from call to \`with\`:
+
+        ${this.payload ? JSON.stringify(this.payload) : '{}'}
+
+        New Payload from call to \`report:\`
+        
+        ${newPayload ? JSON.stringify(newPayload) : '{}'}`
+      );
```

Comment:
Looking over the screenshots you provided, I don't think this is necessary. You're already providing an "after" log below right before the function would be sending a request. 

I think this would result in noise since printing `newPayload` does not provide anything since the user can see what it is in their code. `this.payload` before the merge might be redundant as it will match any previous call to `with()`

Reply:
Agreed